src/backend/distributed/utils/metadata_cache.c

```diff
@@ -933,6 +934,52 @@ HasUninitializedShardInterval(ShardInterval **sortedShardIntervalArray, int shar
 
 
 /*
+ * HasOverlappingShardInterval determines whether the given list of sorted
+ * shards has overlapping ranges.
+ */
+static bool
+HasOverlappingShardInterval(ShardInterval **shardIntervalArray,
```

Comment:
Oh well, I think we'd be better off with the code, but I don't think it's that important. Dropping.